src/drv_oculus_rift/rift.c

```diff
@@ -384,39 +958,114 @@ static ohmd_device* open_device(ohmd_driver* driver, ohmd_device_desc* desc)
 	f = n*10e6;
 	//LOGD("l: %0.3f, r: %0.3f, b: %0.3f, t: %0.3f, n: %0.3f, f: %0.3f", l,r,b,t,n,f);
 	/* eye separation is handled by IPD in the Modelview matrix */
-	omat4x4f_init_frustum(&priv->base.properties.proj_left, l, r, b, t, n, f);
+	omat4x4f_init_frustum(&hmd_dev->base.properties.proj_left, l, r, b, t, n, f);
 	//right eye screen bounds
 	l = -1.0f * priv->display_info.lens_separation/2;
 	r = priv->display_info.h_screen_size/2 - priv->display_info.lens_separation/2;
 	n = priv->display_info.eye_to_screen_distance[1];
 	f = n*10e6;
 	//LOGD("l: %0.3f, r: %0.3f, b: %0.3f, t: %0.3f, n: %0.3f, f: %0.3f", l,r,b,t,n,f);
 	/* eye separation is handled by IPD in the Modelview matrix */
-	omat4x4f_init_frustum(&priv->base.properties.proj_right, l, r, b, t, n, f);
+	omat4x4f_init_frustum(&hmd_dev->base.properties.proj_right, l, r, b, t, n, f);
 
-	priv->base.properties.fov = 2 * atan2f(
+	hmd_dev->base.properties.fov = 2 * atan2f(
 			priv->display_info.h_screen_size/2 - priv->display_info.lens_separation/2,
 			priv->display_info.eye_to_screen_distance[0]);
-
-	// set up device callbacks
-	priv->base.update = update_device;
-	priv->base.close = close_device;
-	priv->base.getf = getf;
+	hmd_dev->id = 0;
+	hmd_dev->hmd = priv;
 
 	// initialize sensor fusion
 	ofusion_init(&priv->sensor_fusion);
 
-	return &priv->base;
+	return priv;
 
 cleanup:
-	if(priv)
-		free(priv);
-
+	if (priv)
+		close_hmd (priv);
 	return NULL;
 }
 
-#define OCULUS_VR_INC_ID 0x2833
-#define SAMSUNG_ELECTRONICS_CO_ID 0x04e8
+static void
+close_hmd (rift_hmd_t *hmd)
+{
+	if (hmd->leds)
+		free (hmd->leds);
+
+	if (hmd->radio_handle)
+		hid_close(hmd->radio_handle);
+	hid_close(hmd->handle);
+	free(hmd);
+}
+
+/* FIXME: This opens the first device that matches the
+ * requested VID/PID/interface, which works fine if there's
+ * 1 rift attached. To support multiple rift, we need to
+ * match parent USB devices like ouvrt does */
+static hid_device* open_hid_dev (ohmd_context* ctx,
+		int vid, int pid, int iface_num)
+{
+	struct hid_device_info* devs = hid_enumerate(vid, pid);
+	struct hid_device_info* cur_dev = devs;
+	hid_device *handle = NULL;
+
+	if(devs == NULL)
+		return NULL;
+
+	while (cur_dev) {
+		if (cur_dev->interface_number == iface_num) {
+			handle = hid_open_path(cur_dev->path);
+			if (handle)
+				break;
+			else {
+				char* path = _hid_to_unix_path(cur_dev->path);
+				ohmd_set_error(ctx, "Could not open %s.\n"
+														"Check your permissions: "
```

Comment:
I can't concentrate on the code when you do this :)

Reply:
Gah! I had so much trouble with tabs vs spaces. Apparently I only ever usually work on projects that use spaces for indent.

Reply:
Much better, thanks.